stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/images/AttachmentGallery.kt

```diff
@@ -9,14 +11,32 @@ import androidx.core.content.res.use
 import androidx.core.view.isVisible
 import androidx.fragment.app.FragmentActivity
 import androidx.viewpager2.widget.ViewPager2
+import com.getstream.sdk.chat.ImageLoader
+import io.getstream.chat.android.core.internal.coroutines.DispatcherProvider
 import io.getstream.chat.android.ui.R
 import io.getstream.chat.android.ui.databinding.StreamUiAttachmentGalleryBinding
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
 
 public class AttachmentGallery : ConstraintLayout {
 
+    private var onSharePictureListener: (pictureUri: Uri) -> Unit = { pictureUri ->
+        ContextCompat.startActivity(
```

Comment:
I'm getting a crash on this on a physical device, seems something is not right with where we place this image / what URI we use for it:

`android.os.FileUriExposedException: file:///storage/emulated/0/Android/data/io.getstream.chat.ui.sample.debug/files/Pictures/share_image_1609768254614.png exposed beyond app through ClipData.Item.getUri()`

Reply:
It seems an issue with the new File API requirement.
Let me update it

Reply:
Yeah, it's gonna be something like that. This was on Android 10, it's also worth testing on Android 11 though.

Reply:
This crash is gone, also tested on Android 11, file permissions look all good now